libs/jacdac-game/debugger.ts

```diff
@@ -50,6 +76,34 @@ namespace jacdac.dbg {
             console.log("");
         }
 
+        makeDiagnostic(label:string, value:string, horizontalOffset: number, color: number, barHeight: number)
+        {
+            const charCount = Math.max(value.length, label.length)
+            let size = this.debugFont.charWidth * charCount + this.marginx
+            screen.fillRect(horizontalOffset,0,size, barHeight,color);
+
+            screen.print(label, (this.marginx / 2) + horizontalOffset, this.marginy, 1, this.debugFont);
+            screen.print(value, (this.marginx / 2) + horizontalOffset, this.debugFont.charHeight + (2 * this.marginy), 1, this.debugFont);
+
+            horizontalOffset += size + 1
+
+            return horizontalOffset;
+        }
+
+        paintDiagnosticsBar(){
+            const diag = jacdac.diagnostics();
```

Comment:
can "diag" be undefined?

Reply:
Yeah, I was planning to fix it up this am ;)